iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectContext.java

```diff
@@ -19,26 +19,71 @@
  */
 package org.sonar.iac.kubernetes.visitors;
 
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.kubernetes.model.ProjectResource;
+
 /**
  * Data class to provide information about the project. This allows to share cross-file knowledge to the individual checks.
  */
 public final class ProjectContext {
+  private final Map<String, Map<String, Set<ProjectResource>>> projectResourcePerNamespacePerPath = new HashMap<>();
 
   private ProjectContext() {
   }
 
+  /**
+   * Get all resources of a given {@code clazz} in a given {@code namespace} and that are accessible to a file with the given {@code path}.
+   * This means that the resources can be in the same file, or in the same directory, or in the descendant directories, but not in the ancestor directories.
+   */
+  public Set<ProjectResource> getProjectResources(String namespace, InputFileContext ctx, Class<? extends ProjectResource> clazz) {
```

Comment:
Please try to use no abbreviations.
For `InputFileContext` I see `ifc` and `ctx` in this PR.